Tasks/AzureRmWebAppDeployment/AzureRMUtil.ts

```diff
@@ -0,0 +1,293 @@
+/// <reference path="../../definitions/node.d.ts" />
+/// <reference path="../../definitions/q.d.ts" />
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+
+var adal = require ('adal-node');
+
+import taskLib = require ('vsts-task-lib/task');
+var request = require ('request');
+import Q = require ('q');
+var parseString = require('xml2js').parseString;
+
+var AuthenticationContext = adal.AuthenticationContext;
+
+function getAuthorizationToken ( SPN ) : Q.Promise <string> {
+
+	var deferred = Q.defer<string>();
+	var authorityUrl = 'https://login.windows.net/' + SPN.tenantID; 
+	var resource = 'https://management.azure.com/';
+
+	var context = new AuthenticationContext (authorityUrl);
+	context.acquireTokenWithClientCredentials (resource, SPN.servicePrincipalClientID, SPN.servicePrincipalKey, function ( error, tokenResponse) {
+		if ( error ) 
+			deferred.reject ( error );
+		else {
+			deferred.resolve ( tokenResponse.accessToken );
+		}
+	});
+
+	return deferred.promise;
+}
+
+exports.getAzureRMWebAppDetails = function ( SPN, webAppName : string ) : Q.Promise<any> {
+
+	var deferred = Q.defer<any> ();
+
+	getAuthorizationToken(SPN)
+	.then(function (accessToken : string) {
+		
+ 		var requestOptions = {
+ 		 	url : 'https://management.azure.com/subscriptions/' + SPN.subscriptionId + '/resources?$filter=resourceType eq \'Microsoft.Web%2FSites\'&api-version=2016-02-01',
+ 		 	auth : {
+ 		 		bearer : accessToken
+ 		 	}
+ 		 };
+
+ 		request (requestOptions, function (error, response, body) {
+ 		 	if ( error ) 
+ 		 		deferred.reject (error);
+ 		 	else {
+ 		 		if ( response.statusCode === 200 ) {
```

Comment:
I think, in TS we can use == instead of ===. Please check.
If possible prefer ==